compiler/src/iree/compiler/Codegen/SPIRV/SPIRVLinkExecutables.cpp

```diff
@@ -40,22 +60,110 @@ struct SPIRVLinkExecutablesPass final
     }
     sourceExecutableOps.resize(retainSize);
 
-    // Guess a module name, if needed, to make the output files readable.
-    std::string moduleName = guessModuleName(moduleOp, "spirv_module");
+    // Note that at runtime, for a particular executable, only one variant of it
+    // will be loaded. So, all variants of an executable are expected to provide
+    // the exact same set of entry points; this way we can guarantee no matter
+    // which variant is chosen, we have all entry points to call into. The same
+    // entry point in different variants may have different target requirements
+    // though.
+    //
+    // The input to the linking stage are a collection of executables, each may
+    // have multiple variants, but only ever provide one entry point. Together
```

Comment:
It's the current status and I think will remain so. With multi-targeting, we are just having more variants in the same executable (think like both SIMT nad cooperative matrix version from matmul). External dispatches are in their own executables so they don't mix together with normal CodeGen executables.

Reply:
Well, we can address it later if that changes in the future. Would it be possible to add a check to bail out on linking given more than 1 entry point? Might help avoid surprises in the future given the assembly for `hal.executable.variant` doesn't explicitly disallow this.

Reply:
Actually, what would happen here if we passed an already linked executable back into this pass? e.g. I do `--compile-to=executable-targets` (or if there is a post-link phase) can I pass it back into iree-compile?

Reply:
If you do that I think lots of the passes would break.. The passes in IREE SPIR-V CodeGen is not meant to support random inputs of all sorts, which is an unbounded space. They are meant to work together.